### DISCUSSION TITLE
Disabling `resultCache` altogether?

### DISCUSSION THREAD

**Discussion by mboudreau** [mikro-orm/mikro-orm#5055] [General]
### Describe the bug

[Following the documentation](https://mikro-orm.io/docs/caching), I found that the results are cached for the ORM, which makes sense, but I would like to disable it outright as I have 2 db queries one after another that needs to get the actual values from the DB and not some cache because of a third party changing data in the db.

I essentially need the latest and most up to date data in this particular scenario.  However, the cache keeps getting in the way, even if I set it to use the `NullCacheAdapter`, or setting global/expiration to `0`, nothing seems to work.  I only see one query being made out of the two and he second find seems to return "out of date" information from what's atually in the DB at the time.  Is there a way to disable this outright?  Either globally or in the config?  My config is using 2 different contextes for 2 different databases, which I'm not sure if this is affecting caching.  We're also using a read replica that has 10ms replication lag, so this shouldn't affect what I'm curently seeing (like 700ms between queries and not seeing a DB hit).  I tried everything and am currently at my wits end.

### Reproduction

Set the configuration to try to disable `resultCache`, like setting `expiration: 0` or `adapter: NullCacheAdapter`.

Then just do a find using the same query twice in a row.  Or even if there's a few hundred milliseconds between calls, it only seem to query the DB once.

### What driver are you using?

@mikro-orm/postgresql

### MikroORM version

6.0.0-rc.3

### Node.js version

18.18.2

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Slack](https://join.slack.com/t/mikroorm/shared_invite/enQtNTM1ODYzMzM4MDk3LWM4ZDExMjU5ZDhmNjA2MmM3MWMwZmExNjhhNDdiYTMwNWM0MGY5ZTE3ZjkyZTMzOWExNDgyYmMzNDE1NDI5NjA).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
Its identity map (or first level cache), not result cache, you can already disable that selectively via find options (either via refresh or via disableIdentityMap option), no plans to do that globally as it completely disables the flush mechanism.

**Reply by mboudreau** [reply]
Ah right, figured it was something I was missing.  Is there any part of the documentation that explains this clearly?  I kinda wasted 2 days trying to figure out why stuff wasn't being updated correctly and couldn't find any documentation on this particular issue when looking for caching. 

Having clear flow diagram of the caching mechanism would be extremely helpful, as well as options.  Does the identity map ever expire or it's help in memory during the whole request (em fork)?

**Reply by B4nan** [reply]
old docs mention it here https://mikro-orm.io/docs/unit-of-work and https://mikro-orm.io/docs/identity-map

new docs mention this in the very first chapter of the getting started guide, and explicitly describe the refreshing mechanism https://mikro-orm.io/docs/next/guide/first-entity#refreshing-loaded-entities
